File: BACK-END/src/main/java/org/kuenteco/backend/dto/auth/ChangePasswordDTO.java
package org.kuenteco.backend.dto.auth;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ChangePasswordDTO {
    @Email
    @NotBlank
    private String email;

    @NotBlank
    private String code;

    @NotBlank
    @Size(min = 8, message = "Password must be at least 8 characters long")
    private String newPassword;

    @NotBlank
    @Size(min = 8, message = "Password must be at least 8 characters long")
    private String confirmNewPassword;
}

File: FRONT-END/lib/dto/auth/request/change_password_dto.dart
class ChangePasswordDTO {
  final String email;
  final String code;
  final String newPassword;

  ChangePasswordDTO({
    required this.email,
    required this.code,
    required this.newPassword,
  });

  factory ChangePasswordDTO.fromJson(Map<String, dynamic> json) {
    return ChangePasswordDTO(
      email: json['email'],
      code: json['code'],
      newPassword: json['newPassword'],
    );
  }

  Map<String, dynamic> toJson() {
    return {'email': email, 'code': code, 'newPassword': newPassword};
  }
}